modules/gatsby/src/components/layout/table-of-contents.jsx

```diff
@@ -183,14 +183,18 @@ const renderRoute = ({route, index, depth, slug, fullyExpanded}) => {
           style={{maxHeight: getHeight(route)}}
         >
           <ul>
-            {route.entries.map((r, idx) =>
-              renderRoute({
+            {route.entries.map((r, idx) => {
+              if (!r.childMarkdownRemark) {
+                console.warn(`Missing content for entry ${idx} in chapter ${route.title}`, route);
```

Comment:
Nit: We use a probe.gl `log` instance in most places. Search for `log.warn`

Reply:
I saw the use of probe's log.warn in logic that I believe is run at "build time" (see process-nodes-json.js), but I see console.warn being used in "run time" components (see example-n.jsx).  Just want to double check that there isn't an explicit decision to keep these logging concepts separate, perhaps to avoid pulling in probe dependencies at runtime?

Reply:
@tgorkin Yes, I see your point. There is a case for having run-time logs. 

Static HTML generation does blur the lines between runtime and build time, but using console inside the React component is probably the right choice..